### ISSUE TITLE
Cannot convert - Problems

### ISSUE THREAD

**Issue by LukasBest42** [ItsClemi/ews-managed-api#15]
Don't know if this is just "me being too dumb to fix it" or an actual issue/bug/whatever. I am working on a project that needs to download attachments from an exchange. It worked with Microsofts "base version" of the api, but i had issues with that needing .NET Framework. I swapped to this(it seems like it is exactly what the doctor ordered), and now my code causes 2 "cannot convert" errors. The relevant code:

ExchangeService service = new(ExchangeVersion.Exchange2013_SP1)
                {
                    Url = new Uri(url),
                    Credentials = new WebCredentials("nope", "not telling")
                };
                **FindItemsResults<Item> results = (FindItemsResults<Item>) service.FindItems(WellKnownFolderName.Inbox, new ItemView(3));**                
                foreach (Item item in results)
                {
                    if (item is EmailMessage)
                    {
                        ItemId id = item.Id;
                        **EmailMessage email = (EmailMessage) EmailMessage.Bind(service, id);**
...

The first bold line throws "CS0029	Cannot implicitly convert type 'System.Threading.Tasks.Task<Microsoft.Exchange.WebServices.Data.FindItemsResults<Microsoft.Exchange.WebServices.Data.Item>>' to 'Microsoft.Exchange.WebServices.Data.FindItemsResults<Microsoft.Exchange.WebServices.Data.Item>' "

The second bold line throws "CS0029 Cannot implicitly convert type 'System.Threading.Tasks.Task<Microsoft.Exchange.WebServices.Data.EmailMessage>' to 'Microsoft.Exchange.WebServices.Data.EmailMessage' "

Explicit converting does not work either from what i can tell
Any help would be greatly appreciated, i've been struggling with my project for months at this point and just want it finished

**Comment by ItsClemi** [comment]
Hi, thanks for reaching out and sorry for the late response. 

You're simply missing the awaits in front of each call. 

Your code compiles fine with:
```csharp
// Required, as the Exchange API collides with the built in Task namespace
using Task = System.Threading.Tasks.Task;

// ...
public class HelloWorld 
{
    public async Task Foo()
    {
        ExchangeService service = new(ExchangeVersion.Exchange2013_SP1)
        {
            Url = new Uri("url"),
            Credentials = new WebCredentials("nope", "not telling")
        };

        var results = await service.FindItems(WellKnownFolderName.Inbox, new ItemView(3));
        foreach (var item in results)
        {
            if (item is EmailMessage)
            {
                ItemId id = item.Id;
                EmailMessage email = await EmailMessage.Bind(service, id);
            }
        }
    }
}
```
You mentioned that you're rewriting .net framework code which may not use async/await at all. Therefore you could also simply unwrap the results by using `GetAwaiter().GetResult()` without introducing major breaking changes into your code base: e.g.

```csharp
        var results = service.FindItems(WellKnownFolderName.Inbox, new ItemView(3)).GetAwaiter().GetResult();
        foreach (var item in results)
        {
            if (item is EmailMessage)
            {
                ItemId id = item.Id;
                EmailMessage email = EmailMessage.Bind(service, id).GetAwaiter().GetResult();
            }
        }
```
But you should consider using async / await through out your code - it's awesome 😄!

**Comment by LukasBest42** [comment]
Thanks, it kinda works? Both alternatives let me find the mail with the .csv attachment I need, except it downloads as an empty file. 
It worked in testing before... bit of an explanation:
I tested it with Microsofts version of the ews api testing from Visual Studio and it worked, despite the project being aimed at net7.0 at the time. It did not work when made the same Code into a .exe ( just crashed instantly on the ExchangeService constructor) due to net /net framework version issues from what i could tell from the error messages. I did not want the hassle of making a new project and changing a bunch of things to work with .net framework 4.whatever, so i am trying this api. now i get empty files...

await/async is also fun because I need to `return` said file's name and therefore need to make other classes async tasks as well...
Still, thanks. Any additional support would be appreciated, but i might be able to figure this out myself? Hopefully?
